.github/workflows/build.yml

```diff
@@ -0,0 +1,178 @@
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: echo "$GITHUB_CONTEXT"
+      -
+        name: Prepare
+        id: prepare
+        run: |
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            echo ::set-output name=tag_name::${GITHUB_REF#refs/tags/}
+          fi
+      -
+        name: Set up Java ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+          java-package: jdk
+      -
+        name: Cache deps
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      -
+        name: Set up Git
+        run: |
+          git config user.name GitHub
+          git config user.email noreply@github.com
+          git remote set-url origin https://x-access-token:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      -
+        name: Release
+        if: success() && github.event_name != 'pull_request' && startsWith(github.ref, 'refs/tags/')
+        run: |
+          ./gradlew clean release \
+            -Prelease.useAutomaticVersion=true \
+            -Prelease.releaseVersion=${{ steps.prepare.outputs.tag_name }} \
+            -x preTagCommit -x createReleaseTag
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      -
+        name: Archive artifact
+        if: success() && matrix.java == '11'
+        uses: actions/upload-artifact@v2
+        with:
+          name: release
+          path: build/libs/*.jar
+      -
+        name: GitHub Release
+        uses: softprops/action-gh-release@v1
+        if: success() && github.event_name != 'pull_request' && startsWith(github.ref, 'refs/tags/')
+        with:
+          draft: true
+          files: |
+            build/libs/*.jar
+          name: ${{ steps.prepare.outputs.tag_name }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  sonar:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java:
+          - 11
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      -
+        name: Set up Java ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+          java-package: jdk
+      -
+        name: Cache deps
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      -
+        name: Sonar
+        run: |
+          ./gradlew sonarqube \
+            -Dsonar.host.url=https://sonarcloud.io \
+            -Dsonar.organization=${SONAR_ORG_KEY} \
+            -Dsonar.projectKey=${SONAR_PROJECT_KEY}
```

Comment:
I've not tried executing this, but does it not need to specify `-Dsonar.pullrequest.*` or `-Dsonar.branch.*` to prevent reporting of changes against the master branch?

Reply:
@mc1arke SonarQube Gradle plugin takes of it like Maven one or I miss something

Reply:
Not that I'm aware of, but I'll check. As far as I was aware, it always treats the current branch as the primary branch (normally master) unless you explicitly tell it otherwise.

Reply:
@mc1arke Ok I will check this out too. We can make some tests [on my fork](https://github.com/crazy-max/sonarqube-community-branch-plugin) if you want, I have enabled Sonarcloud on it: https://sonarcloud.io/dashboard?id=crazy-max_sonarqube-community-branch-plugin